receiver/vmwarevcenterreceiver/client.go

```diff
@@ -0,0 +1,208 @@
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"time"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/performance"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vim25"
+	vt "github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/govmomi/vsan"
+	"github.com/vmware/govmomi/vsan/types"
+)
+
+// vmwareVcenterClient is a client that
+type vmwareVcenterClient struct {
+	moClient   *govmomi.Client
+	vimDriver  *vim25.Client
+	vsanDriver *vsan.Client
+	finder     *find.Finder
+	pc         *property.Collector
+	cfg        *Config
+}
+
+func newVmwarevcenterClient(c *Config) *vmwareVcenterClient {
+	return &vmwareVcenterClient{
+		cfg: c,
+	}
+}
+
+func (vc *vmwareVcenterClient) Connect(ctx context.Context) error {
+	if vc.moClient == nil {
+		sdkURL, err := vc.cfg.SDKUrl()
+		if err != nil {
+			return err
+		}
+		client, err := govmomi.NewClient(ctx, sdkURL, vc.cfg.MetricsConfig.Insecure)
+		if err != nil {
+			return err
+		}
+		tlsCfg, err := vc.cfg.MetricsConfig.LoadTLSConfig()
+		if err != nil {
+			return err
+		}
+		client.DefaultTransport().TLSClientConfig = tlsCfg
+		user := url.UserPassword(vc.cfg.MetricsConfig.Username, vc.cfg.MetricsConfig.Password)
+		err = client.Login(ctx, user)
+		if err != nil {
+			return fmt.Errorf("unable to login to vcenter sdk: %w", err)
+		}
+		vc.moClient = client
+		vc.vimDriver = client.Client
+		vc.pc = property.DefaultCollector(vc.vimDriver)
+		vc.finder = find.NewFinder(vc.vimDriver)
+	}
+	return nil
+}
+
+func (vc *vmwareVcenterClient) Disconnect(ctx context.Context) error {
+	if vc.moClient != nil {
+		return vc.moClient.Logout(ctx)
+	}
+	return nil
+}
+
+func (vc *vmwareVcenterClient) ConnectVSAN(ctx context.Context) error {
+	vsanDriver, err := vsan.NewClient(ctx, vc.vimDriver)
+	if err != nil {
+		return err
+	}
+	vc.vsanDriver = vsanDriver
+	return nil
+}
+
+func (vc *vmwareVcenterClient) Clusters(ctx context.Context) ([]*object.ClusterComputeResource, error) {
+	clusters, err := vc.finder.ClusterComputeResourceList(ctx, "*")
+	if err != nil {
+		return []*object.ClusterComputeResource{}, err
+	}
+	return clusters, nil
+}
+
+func (vc *vmwareVcenterClient) ResourcePools(ctx context.Context) ([]*object.ResourcePool, error) {
+	rps, err := vc.finder.ResourcePoolList(ctx, "*")
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve resource pools: %w", err)
+	}
+	return rps, err
+}
+
+func (vc *vmwareVcenterClient) VMs(ctx context.Context) ([]*object.VirtualMachine, error) {
+	rps, err := vc.finder.VirtualMachineList(ctx, "*")
+	if err != nil {
+		return nil, fmt.Errorf("unable to retrieve resource pools: %w", err)
+	}
+	return rps, err
+}
+
+func (vc *vmwareVcenterClient) CollectVSANCluster(ctx context.Context, clusterRef *vt.ManagedObjectReference, startTime time.Time, endTime time.Time) (*[]types.VsanPerfEntityMetricCSV, error) {
```

Comment:
Since the name and signature are so long, it seems ok to remove `Collect` from the name. It's understood that this is want the client is for in the context of a scraper.

Reply:
Also, we should be consistent with other long signatures and break it into multiple lines.